fortinet-fortiproxy/operations.py

```diff
@@ -0,0 +1,399 @@
+from .constant import *
+
+logger = get_logger('fortinet-fortiproxy')
+
+
+class FortiProxy(object):
+    def __init__(self, config, *args, **kwargs):
+        self.apikey = config.get('api_key')
+        url = config.get('server_url').strip('/')
+        if not url.startswith('https://') and not url.startswith('http://'):
+            self.url = 'https://{0}/api/v2/'.format(url)
+        else:
+            self.url = url + '/api/v2/'
+        self.verify_ssl = config.get('verify_ssl')
+
+    def make_rest_call(self, url, method, data=None, params=None):
+        try:
+            url = self.url + url
+            params.update({'access_token': self.apikey})
+            headers = {
+                'Accept': 'application/json'
+            }
+            logger.debug("Endpoint {0}".format(url))
+            response = requests.request(method, url, data=data, params=params, headers=headers, verify=self.verify_ssl)
+            logger.debug("response_content {0}:{1}".format(response.status_code, response.content))
+            if response.ok or response.status_code == 204:
+                logger.info('Successfully got response for url {0}'.format(url))
+                if 'json' in str(response.headers):
+                    return response.json()
+                else:
+                    return response
+            elif response.status_code == 404:
+                return response.json()
+            else:
+                raise ConnectorError("{0}".format(response.content))
+        except requests.exceptions.SSLError:
+            raise ConnectorError('SSL certificate validation failed')
+        except requests.exceptions.ConnectTimeout:
+            raise ConnectorError('The request timed out while trying to connect to the server')
+        except requests.exceptions.ReadTimeout:
+            raise ConnectorError(
+                'The server did not send any data in the allotted amount of time')
+        except requests.exceptions.ConnectionError:
+            raise ConnectorError('Invalid Credentials')
+        except Exception as err:
+            raise ConnectorError(str(err))
+
+
+def check_payload(payload):
+    updated_payload = {}
+    for key, value in payload.items():
+        if isinstance(value, dict):
+            nested = check_payload(value)
+            if len(nested.keys()) > 0:
+                updated_payload[key] = nested
+        elif value:
+            updated_payload[key] = value
+    return updated_payload
+
+
+def create_firewall_policy(config, params):
+    fp = FortiProxy(config)
+    endpoint = 'cmdb/firewall/policy'
+    query_parameter = {
+        'vdom': params.pop('vdom', ''),
+        'action': params.pop('action', ''),
+        'nkey': params.pop('nkey', '')
+    }
+    params.update({'type': Policy_Type.get(params.get('type'))}) if params.get('type') else ''
+    params.update({'action': params.get('action').lower()}) if params.get('action') else ''
```

Comment:
the if condition will always be false here as you are doing params.pop('action', '') above this